fix: relocate Automations snippet includes and fix localized imports (#2727)
wandb/docs: models/sweeps/start-sweep-agents.mdx

---
description: Start or stop a W&B sweep agent on one or more machines.
title: Start a sweep agent
keywords: [wandb agent, sweep ID, hyperparameter search, run count, signal forwarding]
---

This page explains how to start a sweep agent on one or more machines so that W&B can run your hyperparameter search. Sweep agents use the sweep configuration that you define when you [initialize a sweep](/models/sweeps/initialize-sweeps) to explore different hyperparameter combinations. W&B creates a new run for each hyperparameter combination that the sweep agent runs.

> After your agents are running, see [Manage sweeps](/models/sweeps/pause-resume-and-cancel-sweeps) to learn how to pause, resume, stop, or cancel a sweep.

Reviewer: ```suggestion
See [Manage sweeps](/models/sweeps/pause-resume-and-cancel-sweeps) to learn how to pause, resume, stop, or cancel a running sweep agent.
```

Author: Applied.

> Before you continue, make sure you:

Reviewer: ```suggestion
Before you continue:
```

Author: Applied.

> * Have configured your training script to create and track hyperparameter combinations with W&B. For more information, see [Add W&B to your code](./add-w-and-b-to-your-code#training-script-with-w%26b-python-sdk).

Reviewer: ```suggestion
* Configure your training script to create and track hyperparameter combinations with W&B. See [Add W&B to your code](./add-w-and-b-to-your-code#training-script-with-w%26b-python-sdk) for examples.
```

Author: Applied, but made the link absolute and corrected the anchor to `#python-script-or-notebook` (the `#training-script-…` anchor doesn't exist; this matches the working link later on the page).

> * Have a [configuration file](./define-sweep-configuration) defined for your sweep.

Reviewer: ```suggestion
* Define a [configuration file](./define-sweep-configuration) for your sweep.
```
These weird participle patterns seem AI-sloppy.

Author: Applied (absolute link).

> You can start a sweep agent with the W&B CLI or from within a Jupyter notebook or Python script. The following code snippets demonstrate both methods. For either method, provide the sweep ID that W&B returns when you initialize the sweep. The sweep ID has the form:

Reviewer: ```suggestion
These examples show how to start a sweep agent with the W&B CLI or from Python. You need the sweep ID that W&B returns when you initialize the sweep. The sweep ID has the form:
```

Author: Applied.
